sdk/metric/meter.go

```diff
@@ -104,20 +104,44 @@ func (m *meter) Int64Histogram(name string, options ...metric.Int64HistogramOpti
 	return i, validateInstrumentName(name)
 }
 
+// int64ObservableInstrument returns a new observable identified by the Instrument.
+// It registers callbacks for each reader's pipeline.
+func (m *meter) int64ObservableInstrument(id Instrument, callbacks []metric.Int64Callback) (int64Observable, error) {
+	inst := newInt64Observable(m, id.Kind, id.Name, id.Description, id.Unit)
+	for i, pipe := range m.pipes {
+		// Connect the measure functions for instruments in this pipeline with the
+		// callbacks for this pipeline.
+		inserter := m.int64Resolver.inserters[i]
```

Comment:
What do you mean by fragile? Does the order matters?

Reply:
If, somehow, the ordering of inserters or pipelines changed, it would break the SDK in bizarre ways, since collect on each reader will collect observations for other reader's instruments.

Reply:
What about adding a `Callback` method to the `Inserter` that will insert into the pipeline it holds?

Reply:
done